apps/game_backend/lib/game_backend/curse_of_mirra/matches.ex

```diff
@@ -55,7 +56,15 @@ defmodule GameBackend.CurseOfMirra.Matches do
         Enum.find(user.units, fn unit -> unit.character.name == result["character"] end)
         |> case do
           nil ->
-            {:error, :unit_not_found}
+            unit_params =
+              Units.get_unit_default_values(result["character"])
+              |> Map.put(:user_id, user.id)
+
+            reward = match_prestige_reward(unit_params, result["position"], prestige_config[:rewards])
+            changes = calculate_rank_and_amount_changes(unit_params, reward, prestige_config[:ranks])
+
+            Map.merge(unit_params, changes)
+            |> Units.insert_unit()
```

Comment:
I think this is bad for several reasons:
1. The function is called "give_prestige" and it inserts you a unit if you don't have it, which is way more than giving prestige (beyond responsibility).
2. The {:error, :unit_not_found} was intended. What happens if a user lose a unit due to some bug? We'll never notice since we're always inserting it here over and over.
3. Players shouldn't be allowed to play with a Unit they don't have. If I don't have some character unlocked and I manage to play with it somehow (bug), I'll be rewarded with it.

Reply:
1. Fair point, maybe moving this to a different multi operation is better
2. 3. We don't have an instance of "Log in" or "Character unlock" to assign this units right now, they're just broken. Maybe this may work as a patch until we develop those, i can create an issue and will be worked in the future when we have time.

Reply:
Due to our need to fix this we should:
1. Move this whole "patch" to another function. Encapsulating this behavior will help us in the future to remove it.
2. Create an issue to reimplement this properly. Also, attach its link in the code.